src/Dialect/ONNX/MLIRDialectBuilder.cpp

```diff
@@ -392,6 +392,12 @@ memref::CastOp MemRefBuilder::cast(Value input, MemRefType outputType) const {
 }
 
 Value MemRefBuilder::dim(Value val, int64_t index) const {
+  assert((val.getType().isa<MemRefType>() ||
+             val.getType().isa<UnrankedMemRefType>()) &&
+         "memref::DimOp expects input operand to have MemRefType or "
+         "UnrankedMemRefType");
+  assert(index >= 0 && "Expecting a vaid index");
```

Comment:
Typo: vaid -> valid

Reply:
> @etiotto thanks for this amazing investigation and fixes of bugs to integrate the latest LLVM, impressive.

You are welcome, it was a good learning experience.